### PR TITLE
Update organization id

### PR DESCRIPTION
**Pull request by BalmungSan**
This PR updates the organization id so we can do the following `0.18.0` release under `io.github.neotypes`

It also adds the **Changelog** for that release.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3,7 +3,7 @@
 > neotype - a type specimen that is selected subsequent to the description of a species to replace a preexisting type that has been lost or destroyed.
 
 ![CI](https://github.com/neotypes/neotypes/workflows/CI/badge.svg?branch=main)
-[![Maven Central](https://img.shields.io/maven-central/v/com.dimafeng/neotypes_2.12.svg)](https://maven-badges.herokuapp.com/maven-central/com.dimafeng/neotypes_2.12)
+[![Maven Central](https://img.shields.io/maven-central/v/io.github.neotypes/neotypes-core_2.12.svg)](https://maven-badges.herokuapp.com/maven-central/io.github.neotypes/neotypes-core_2.12)
```
Comment (BalmungSan) [inline README.md:6]: I am not 100% sure if this will be the new URL of the badge.

File: build.sbt [hunk 2]
```diff
@@ -51,7 +51,7 @@ ThisBuild / scmInfo ~= {
 // Global settings.
 ThisBuild / scalaVersion := "2.12.14"
 ThisBuild / crossScalaVersions := Seq("2.12.14", "2.13.6")
-ThisBuild / organization := "com.dimafeng"
+ThisBuild / organization := "io.github.neotypes"
```
Comment (BalmungSan) [inline build.sbt:54]: Do we also need to make some changes to [the `publishTo` settings](https://github.com/neotypes/neotypes/pull/359/files#diff-5634c415cd8c8504fdb973a3ed092300b43c4b8fc1e184f7249eb29a55511f91R73-R83)?

File: site/src/main/mdoc/index.md [hunk 3]
```diff
@@ -10,7 +10,7 @@ position: 0
 > neotype - a type specimen that is selected subsequent to the description of a species to replace a preexisting type that has been lost or destroyed.
 
 [![Build Status](https://travis-ci.org/neotypes/neotypes.svg?branch=main)](https://travis-ci.org/neotypes/neotypes)
-[![Maven Central](https://img.shields.io/maven-central/v/com.dimafeng/neotypes_2.12.svg)](https://maven-badges.herokuapp.com/maven-central/com.dimafeng/neotypes_2.12)
+[![Maven Central](https://img.shields.io/maven-central/v/io.github.neotypes/neotypes-core_2.12.svg)](https://maven-badges.herokuapp.com/maven-central/io.github.neotypes/neotypes-core_2.12)
```
Comment (BalmungSan) [inline site/src/main/mdoc/index.md:13]: Ditto badge url.

**Review by BalmungSan — COMMENTED** [review]
@dimafeng I will push the release note shortly, after that all you need to do is change the release date and do the release whenever you can.

Let me know if there is anything else I may do to help!

File: site/src/main/mdoc/changelog.md [hunk 4]
```diff
@@ -7,6 +7,57 @@ position: 100
 
 # Changelog
 
+## v0.18.0 _(2021-06-xx)_
```
Comment (BalmungSan) [inline site/src/main/mdoc/changelog.md:10]: Remember to update this value before the release!

File: README.md (hunk 1 above)
Comment (dimafeng) [reply README.md:6]: I'm afraid that the only way to check it is to release the version 😁

File: build.sbt (hunk 2 above)
Comment (dimafeng) [reply build.sbt:54]: For some reason your link is not working. I'll figure out what else needs to be changed during release

**Comment by dimafeng** [comment]
@BalmungSan seems something is wrong with site snippets (the build is failing). Could you please take a look?

**Pull request merged**